src/txmempool.cpp

```diff
@@ -634,17 +634,20 @@ void CTxMemPool::removeForBlock(const std::vector<CTransactionRef>& vtx, unsigne
     }
     // Before the txs in the new block have been removed from the mempool, update policy estimates
     if (minerPolicyEstimator) {minerPolicyEstimator->processBlock(nBlockHeight, entries);}
+    std::vector<CTransactionRef> txs_removed_for_block;
```

Comment:
In  [fc77cac](https://github.com/willcl-ark/bitcoin/pull/7/commits/fc77cac24dcb31dd944e7693be7a6ed49b16dcc5), commit I am deleting the`std::vector<const CTxMemPoolEntry*> entries;` and the code that appends it values, after I modify `CBlockPolicyEstimator` `processBlock` to not need vector of `CTxMempoolEntry`.

One reason I think we might not want to pass vector of `CTxMempoolEntry` is making copies of  `CTxMempoolEntry` and pass the reference to the vector might be a bit inefficient.

Expect maybe we can pass `std::vector<const CTxMemPoolEntry*> entries;`  but this not blocking and the mempool erasing the entry will that not affect the pointers?

Reply:
>edit: is this so that our "parallel mempool" in fee estimator is as lightweight as possible?

for the paralled mempool `CTxMempoolEntry` might even be more desirable because it has ancestors info's.
`CTransationRef` could also work, but I did not think this through.